clients/python/src/model_registry/_client.py

```diff
@@ -17,27 +19,50 @@ class ModelRegistry:
     def __init__(
         self,
         server_address: str,
-        port: int,
+        port: int = 443,
+        *,
         author: str,
-        client_key: str | None = None,
-        server_cert: str | None = None,
-        custom_ca: str | None = None,
+        user_token: bytes | None = None,
+        custom_ca: bytes | None = None,
     ):
         """Constructor.
 
         Args:
             server_address: Server address.
-            port: Server port.
+            port: Server port. Defaults to 443.
+
+        Keyword Args:
             author: Name of the author.
-            client_key: The PEM-encoded private key as a byte string.
-            server_cert: The PEM-encoded certificate as a byte string.
-            custom_ca: The PEM-encoded root certificates as a byte string.
+            user_token: The PEM-encoded user token as a byte string. Defaults to content of path on envvar KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: The PEM-encoded root certificates as a byte string. Defaults to contents of path on envvar CERT.
         """
         # TODO: get args from env
         self._author = author
-        self._api = ModelRegistryAPIClient(
-            server_address, port, client_key, server_cert, custom_ca
-        )
+
+        if not user_token:
+            # /var/run/secrets/kubernetes.io/serviceaccount/token
+            sa_token = os.environ.get("KF_PIPELINES_SA_TOKEN_PATH")
+            if sa_token:
+                user_token = Path(sa_token).read_bytes()
+            else:
+                warn("User access token is missing", stacklevel=2)
+
+        root_ca = None
+        if not custom_ca and port == 443:
```

Comment:
why are you hard coding `port == 443` is secure, 443 is just a default port that supports encrypted channels. it is user driven it can be anything like 1443. so not hard code anything.

also what does "secure" mean in your code, is it having an encrypted channel considered here as secure or user authentication as secure, or a combination of both? make that intent clear in your code. imo, when ca is given it a `encrypted` channel. user-token should be available both ways, as we never leave the connection not authenticated. if user-token we should just fail. @dhirajsb